src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,116 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .mapToDouble(employee -> employee.getPerson().getAge())
+                .average().orElse(-1);
+
+        assertThat(expected, Matchers.closeTo(33.66, 0.1));
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt(p -> p.getFullName().length()))
+                .orElseThrow(UnsupportedOperationException::new);
+
+        assertThat(expected, Matchers.is(employees.get(1).getPerson()));
+    }
+
+    @Test
+    void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        Employee expected = employees.stream()
+                .max(Comparator.comparingInt(employee -> employee.getJobHistory().stream()
+                    .map(JobHistoryEntry::getDuration)
+                    .max(Integer::compareTo).orElse(-1))).orElseThrow(UnsupportedOperationException::new);
+
+        assertThat(expected, Matchers.is(employees.get(4)));
+    }
+
+    /**
+     * Вычислить общую сумму заработной платы для сотрудников.
+     * Базовая ставка каждого сотрудника составляет 75_000.
+     * Если на текущей позиции (последняя в списке) он работает больше трех лет - ставка увеличивается на 20%
+     */
+    @Test
+    void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getJobHistory)
+                .map(positions -> positions.get(positions.size() - 1))
+                .map(JobHistoryEntry::getDuration)
+                .map(d -> d > 3 ? 75000d * 1.2 : 75000)
```

Comment:
Here u can use `mapToDouble(d -> d > 3 ? 75000d * 1.2 : 75000d)` and just delete the next line